### ISSUE TITLE
Type-only imports trigger no-shadow despite ignoreTypeValueShadow: true

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1226]
## Summary

Reproduced with @utoo/lint 0.4.1 on Node.js 22.23.1, macOS arm64. The imported name belongs only to the type namespace. With ignoreTypeValueShadow: true, a local value with the same name should be allowed.

## Minimal reproduction

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0
```

`example.tsx`:

```tsx
import type { Item } from "./types"; export function example() { const Item: Item={value:1};return Item; }
```

`types.ts`:

```ts
export interface Item {value:number;}
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/no-shadow": [
      "error",
      {
        "ignoreTypeValueShadow": true
      }
    ]
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/no-shadow": [
      "error",
      {
        "ignoreTypeValueShadow": true
      }
    ]
  }
};
```

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Result

ESLint reports no diagnostics. utlint reports one @typescript-eslint/no-shadow error for Item.

Both tools check the same file successfully; there are no syntax errors or lint-disable comments.

**Comment by fireairforce** [comment]
Retested with @utoo/lint 0.4.2, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0 on Node.js 22.23.1 (macOS arm64).

The original explicit `ignoreTypeValueShadow: true` reproduction is fixed. A default-option discrepancy remains with the same source files and setup:

| Rule value | ESLint diagnostics | utlint diagnostics |
|---|---:|---:|
| `"error"` | 0 | 1 |
| `["error", {}]` | 0 | 1 |
| `["error", {"ignoreTypeValueShadow": true}]` | 0 | 0 |
| `["error", {"ignoreTypeValueShadow": false}]` | 1 | 1 |

For the first two configurations, utlint still reports `'Item' is already declared in the upper scope on line 1 column 15.` Omitting the option should behave like the reference rule's default (`true`). No other rules are enabled.
